src/game/shared/neo/neo_player_shared.cpp

```diff
@@ -157,9 +161,43 @@ void CheckPingButton(CNEO_Player* player)
 		{
 			player->m_flNextPingTime = gpGlobals->curtime;
 		}
+
+		UpdatePingCommands(player, tr.endpos);
 	}
 }
 
+ConVar sv_neo_bot_cmdr_enable("sv_neo_bot_cmdr_enable", "0",
+	FCVAR_CHEAT | FCVAR_REPLICATED, "Allow bots to follow you after you press use on them", true, 0, true, 1);
+
+#ifdef GAME_DLL
+static ConVar sv_neo_bot_cmdr_stop_distance_sq_max("sv_neo_bot_cmdr_stop_distance_sq_max", "50000",
+	FCVAR_CHEAT, "Maximum distance bot following gap interval can be set by player pings", true, 5000, true, 500000);
+static ConVar sv_neo_bot_cmdr_ping_ignore_delay_min_sec("sv_neo_bot_cmdr_ping_ignore_delay_min_sec", "3",
+	FCVAR_CHEAT, "Minimum time bots ignore pings for new waypoint settings", true, 0, true, 1000);
+#endif // GAME_DLL
+
+void UpdatePingCommands(CNEO_Player* player, const Vector& pingPos)
+{
+#ifdef GAME_DLL
+	if (sv_neo_bot_cmdr_enable.GetBool())
+	{
+		player->m_vLastPingByStar.GetForModify(player->GetStar()) = pingPos;
+		float distSqrToPing = player->GetAbsOrigin().DistToSqr(pingPos);
+		if (distSqrToPing < sv_neo_bot_cmdr_stop_distance_sq_max.GetFloat())
+		{
+			// If pinging close to self, calibrate follow distance of commanded bots based on distance to ping
+			player->m_flBotDynamicFollowDistanceSq = Clamp(distSqrToPing, 5000.0f, sv_neo_bot_cmdr_stop_distance_sq_max.GetFloat());
```

Comment:
Nit: maybe it'd be good to get the clamp min value from the cvar directly, so these cannot go out of sync if someone updates the cvar min value code but forgets about this hard-coded 5000.f